test/unit/output/pudltabl_test.py

```diff
@@ -0,0 +1,130 @@
+        """Test table name substitution for monthly aggregation."""
+        pudl_tabl = PudlTabl(freq="MS")
+
+        result = pudl_tabl._agg_table_name("out_eia923__AGG_fuel_receipts")
+        assert result == "out_eia923__monthly_fuel_receipts"
+
+    def test_no_aggregation_removes_agg(self):
+        """Test that _AGG is removed when freq is None."""
+        pudl_tabl = PudlTabl(freq=None)
+
+        result = pudl_tabl._agg_table_name("out_eia923__AGG_generation")
+        assert result == "out_eia923__generation"
+
+    def test_non_aggregated_table_unchanged(self):
+        """Test that non-aggregated table names are unchanged."""
+        pudl_tabl = PudlTabl(freq="YS")
+
+        result = pudl_tabl._agg_table_name("core_eia860__scd_generators")
+        assert result == "core_eia860__scd_generators"
+
+
+class TestPudlTablInitialization:
+    """Test PudlTabl initialization and validation."""
+
+    def test_valid_frequencies(self):
+        """Test that valid frequencies are accepted."""
+        # These should all work without raising exceptions
+        PudlTabl(freq=None)
+        PudlTabl(freq="YS")
+        PudlTabl(freq="MS")
+
+    def test_invalid_frequency_raises_error(self):
+        """Test that invalid frequencies raise ValueError."""
+        with pytest.raises(ValueError, match="freq must be one of None, 'MS', or 'YS'"):
+            PudlTabl(freq="QS")  # Quarterly not supported
+
+        with pytest.raises(ValueError, match="freq must be one of None, 'MS', or 'YS'"):
+            PudlTabl(freq="D")  # Daily not supported
+
+        with pytest.raises(ValueError, match="freq must be one of None, 'MS', or 'YS'"):
+            PudlTabl(freq="invalid")
+
+    def test_date_conversion(self):
+        """Test that various date formats are properly converted."""
+        # String dates
+        pudl_tabl = PudlTabl(start_date="2020-01-01", end_date="2022-12-31")
+        assert isinstance(pudl_tabl.start_date, pd.Timestamp)
+        assert isinstance(pudl_tabl.end_date, pd.Timestamp)
+
+        # Date objects
+        pudl_tabl = PudlTabl(start_date=date(2020, 1, 1), end_date=date(2022, 12, 31))
+        assert isinstance(pudl_tabl.start_date, pd.Timestamp)
+        assert isinstance(pudl_tabl.end_date, pd.Timestamp)
+
+        # Datetime objects
+        pudl_tabl = PudlTabl(
+            start_date=datetime(2020, 1, 1, 10, 30),
+            end_date=datetime(2022, 12, 31, 15, 45),
+        )
+        assert isinstance(pudl_tabl.start_date, pd.Timestamp)
+        assert isinstance(pudl_tabl.end_date, pd.Timestamp)
+
+    def test_invalid_date_format_raises_error(self):
+        """Test that invalid date formats raise appropriate errors."""
+        with pytest.raises((ValueError, TypeError)):
+            PudlTabl(start_date="not-a-date")
+
+        with pytest.raises((ValueError, TypeError)):
+            PudlTabl(end_date="2020-13-45")  # Invalid month/day
+
+    def test_fill_net_gen_parameter(self):
```

Comment:
Another instance of "this test seems like it *could* be testing something useful but isn't yet". Currently it's just "does the parameter get set?", but we could be testing "does `fill_net_gen` affect the behavior of `gen_eia923` as expected?"